3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -63,46 +64,91 @@
 
 all_GSEA_results = []
 all_signatures = []
-#negative_control = []
-for col in signature.iloc[:,1:50].columns:
+neg_GSEA_results = []
+negative_control = []
+
+for col in signature.iloc[:,1:101].columns:
     df = signature.iloc[:,[0,int(col)]]
     result = blitz.gsea(df, library)
     all_GSEA_results.append(result.assign(z_dim=f"z_{col}"))
     all_signatures.append(df)
-    #add line here to create a negative control--this will be a random scramble of the gene scores
+    neg_df = df.copy()
+    neg_df[col] = neg_df[col].sample(frac=1).reset_index(drop=True)
+    neg_result = blitz.gsea(neg_df, library)
+    neg_GSEA_results.append(neg_result.assign(z_dim=f"z_{col}"))
+    negative_control.append(neg_df)
+
 all_GSEA_results
+neg_GSEA_results
 
 
 # In[7]:
 
 
 # stack up all of the results to be analyzed
 all_GSEA_results= pd.concat(all_GSEA_results)
+neg_GSEA_results = pd.concat(neg_GSEA_results)
 
 
 # In[8]:
 
 
 # sort by what you want to evaluate
-all_GSEA_results.sort_values(by='es', ascending=False)
+all_GSEA_results['rank'] = (-np.log10(all_GSEA_results.pval))*(all_GSEA_results.es)
+all_GSEA_results.sort_values(by='pval', ascending = True)
+
+neg_GSEA_results['rank'] = (-np.log10(neg_GSEA_results.pval))*(neg_GSEA_results.es)
+neg_GSEA_results.sort_values(by='es', ascending = False)
 
 
 # In[9]:
 
 
-plt.scatter(x=all_GSEA_results['es'],y=all_GSEA_results['pval'].apply(lambda x:-np.log10(x)),s=1)
+plt.figure()
+plt.scatter(x=all_GSEA_results['es'],y=all_GSEA_results['pval'].apply(lambda x:-np.log10(x)),s=10)
+plt.xlabel('log2 Fold Change (ES)')
+plt.ylabel('-log10(pvalue)')
+plt.title('Gene Set Enrichment Analysis')
+
+plt.figure()
+plt.scatter(x=neg_GSEA_results['es'],y=neg_GSEA_results['pval'].apply(lambda x:-np.log10(x)), s=10)
+plt.xlabel('log2 Fold Change (ES)')
+plt.ylabel('-log10(pvalue)')
+plt.title('Control Gene Set Enrichment Analysis')
 
 
 # In[10]:
 
 
-# plot the enrichment results and save to png--this code needs some work done to it in order to work!
-fig = blitz.plot.running_sum(signature, "DIABETIC CARDIOMYOPATHY", library, result=result, compact=False)
-fig.savefig("running_sum.png", bbox_inches='tight')
+# Using VAE generated data
+
+z = 0
+for df in all_signatures:
+    
+    z = z+1
+
+    fig = blitz.plot.running_sum(df, "regulation of transcription from RNA polymerase II promoter in response to hypoxia (GO:0061418)", library, result=result, compact=False)
```

Comment:
Why is this label hardcoded? Won't it change each time? Is there a way to not hardcode?

Reply:
(PR comments should involve a discussion - it helps the reader to understand decisions that you as the developer made. You as the developer also benefit from multiple eyes on the PR, and, as the developer, you have final say.)